test/models/user_test.rb

```diff
@@ -0,0 +1,23 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
```

Comment:
Let us add tests for the user's association with the other models. For example one of the tests we wanna do is if indeed user `has_many` friends. The same for the other associations in your model.

Special mention to the bug I told you about in the `user.rb`. We need to catch that in the tests. Also let's test for the deletion of a user's posts, comments, friendrequests, friendships should the user be destroyed